server/hedley/modules/custom/hedley_event/hedley_event.module

```diff
@@ -40,52 +38,74 @@ function hedley_event_get_events_by_group($group, $language_code, $show_as_block
-      // Date is required but just in case date is missing in some translation.
-      continue;
-    }
+/**
+ * Get the elm element for an Event.
+ *
+ * @param object $event
+ *   The event.
+ * @param string $language_code
+ *   The language code (e.g. "he", "ar").
+ * @param string $image_style
+ *   The image style for the event image, defaults to large.
+ *
+ * @return array|bool
+ *   The event item or nothing if something is missing.
+ */
+function hedley_event_get_event_item($event, $language_code, $image_style = 'large') {
+  global $user;
 
-    $location_unfiltered = $wrapper->field_location->value();
-    // Keep only "url" and "title" sub properties and remove any value that is
-    // missing one of those properties, although they are required but just in
-    // case.
-    $location = array_filter(['url' => $location_unfiltered['url'], 'title' => $location_unfiltered['title']], function ($value) {
-      return !is_null($value);
-    });
+  $wrapper = entity_metadata_wrapper('node', $event);
+  $wrapper->language($language_code);
 
-    $event = [
-      'id' => $wrapper->getIdentifier(),
-      'name' => check_plain($wrapper->label()),
-      'image_url' => !empty($image) ? image_style_url('square_large_image', $image['uri']) : hedley_municipality_get_default_image($wrapper->getBundle()),
-      'description' => !empty($wrapper->body->value()) ? $wrapper->body->value->value() : NULL,
-      'date' => hedley_event_timestamp_to_iso8601($date['value']),
-      // Make sure the end date isn't the same as the start date.
-      'end_date' => !empty($date) && $date['value2'] && $date['value'] != $date['value2'] ? hedley_event_timestamp_to_iso8601($date['value2']) : NULL,
-      'recurring_weekly' => $wrapper->field_recurring_weekly->value(),
-      'ticket_price' => $wrapper->field_ticket_price->value() ? $wrapper->field_ticket_price->value() : NULL,
-      'location' => !empty($location) ? $location : NULL,
-      'showEditLink' => og_node_access($node, 'update', $user) == NODE_ACCESS_ALLOW,
-    ];
+  // Ignore events without a title in the current language.
+  if (!$wrapper->label()) {
+    return FALSE;
+  }
+
+  $image = $wrapper->field_image->value();
+  $date = $wrapper->field_date->value();
 
-    // Remove empty values (elm expects the correct value or nothing), but keep
-    // the values with `FALSE` value.
-    $events[] = array_filter($event, function ($value) {
-      return !is_null($value);
-    });
+  if (empty($date)) {
+    // Date is required but just in case date is missing in some translation.
+    return FALSE;
   }
 
-  return $events;
+  $location_unfiltered = $wrapper->field_location->value();
+  // Keep only "url" and "title" sub properties and remove any value that is
+  // missing one of those properties, although they are required but just in
+  // case.
+  $location = array_filter(['url' => $location_unfiltered['url'], 'title' => $location_unfiltered['title']], function ($value) {
```

Comment:
I'm not clear on this filtering and the one below, can you please clarify.

Reply:
So basically this will remove any property in the `location` that is not either `url` or `title` and at the same time will remove those properties if their values are `null`.
The one below will remove any `null` value from the `array`.

Do you want me to try and update the comment if it's not clear?

Reply:
Ok, leave as-is.